Update anchor-lang and anchor-spl to the latest version. (starter branch)

wuuer opened the pull request:
Use InitSpace to calculate space needed for accounts. 
Replace "@project-serum/anchor" with "@coral-xyz/anchor" 
Use the latest "connection.confirmTransaction()"
Delete Unnecessary parameters are found in the test typescript. 
add "insecure_withdraw" test case in the test typescript.

Commit:
Update anchor-lang and anchor-spl to the latest version.
Use InitSpace to calculate space needed for accounts.
Replace "@project-serum/anchor" with "@coral-xyz/anchor"
Use the latest "connection.confirmTransaction()"
Delete Unnecessary parameters are found in the test typescript.
add "insecure_withdraw" test case in the test typescript.

wuuer committed:
Merge branch 'starter' into starter

mikemaccana commented:
I've moved this code to https://github.com/solana-developers/owner-checks/ - note another set of updates have been merged so if this goes in it will need to be updated to handle the conflicts.

wuuer commented:
> I've moved this code to https://github.com/solana-developers/owner-checks/ - note another set of updates have been merged so if this goes in it will need to be updated to handle the conflicts.

Thanks. I replaced the repository link with this in owner-checks.md
